# Microbial Ecology {#sec-microbial-ecology}

This section introduces key concepts of how we use molecular techniques to identify and quantify the composition of microbial communities, to study microbial ecology.

## What is microbial ecology?

Microbial ecology is the scientific study of natural microbial communities. It is a hugely important, but arguably underdeveloped area of biology. Many important questions remain to be completely answered in detail:

- the composition of microbial communities
  - what microbes are found in a community?
  - what proportion of each microbe species (or other group) is present in a community?
  - what are the physical limits of the community?
- interactions within the community
  - is the community _stable_ (or is it growing, shrinking, or changing composition)?
  - do community members share or compete for resources, and which resources?
  - do the community members effectively operate as a larger-scale system?
- interactions of a community with other organisms
  - does the gut microbial community influence the host animal's nutrition, health, or disease status?
  - does the rhizosphere microbial community influence a plant's ability to extract nutrition or energy from the environment, and contribute to protection against pathogens?
- interactions between communities
  - when two communities come together, what does the resulting community look like?
  - are there two or more separate (or interacting communities), or do they somehow combine?
- engineering new microbial communities
  - can we predict which microbes can combine into a productive community?
  - can we choose these microbes to engineer a community that achieves a particular goal (e.g. plant protection or gut health)
  - can we perturb existing communities to make them beneficial to health or achieve some other goal?

This array of questions can be summarised more simply as [@Prosser2020-aw]:

1. Who is there?
2. What is meant by "there"?
3. Who is doing what in there?
4. What is the effect of doing â€¦ to "there"?

## Answering questions about microbial ecology using molecular techniques

Molecular techniques enabled by modern sequencing approaches, such as amplicon marker sequencing (e.g. 16S, ITS1 methods) - also known as _metabarcoding_ - and sequencing the entire genomic DNA of a complex sample - also known as _metagenomics_ - have become the dominant approach to understanding microbial community composition [@Boughner2016-vx]. These tools give rapid insights into the complex composition of microbial communities that are unattainable using laborious, time-consuming culture-based techniques.

::: {.callout-tip}
The two words are quite similar, but it is important to distinguish between

- **metabarcoding**: sequencing a single marker sequence from a population
- **metagenomics**: deep sequencing of the entire genomic DNA material in a sample, often attempting to assemble representative genomes from the data
:::

In addition to their speed and the scale of data they can produce, these molecular techniques have additional advantages:

- The (sequence) data collected can be preserved and shared, alongside metadata describing the experiment, in public databases. This enables reanalysis and integration into larger-scale studies.
- Fastidious organisms, or those that might be outcompeted in a laboratory culture, can be sequenced and identified.
- Organisms present in low numbers can be sequenced and identified.

These molecular techniques answer the question of _"Who is there?"_, in two ways:

1. They tell us what kinds of organism are present (often to _species_ or _genus_ level)
2. They give us a relative count of how much of each kind of organism is present

Taken together, these answers tell us about the _diversity_ of the community.

::: {.callout-note}
We will introduce measures of community diversity in [a later section](diversity_measures.qmd).
:::


### Which organisms are present?

The two approaches of _metabarcoding_ and _metagenomics_ can each tell us which organisms are present in a sample.

In _metabarcoding_, we use the polymerase chain reaction (PCR) and primers _targeted_ specifically to amplify a gene fragment or other stretch of DNA that we believe to be present in all organisms of interest. For bacteria, this is usually a fragment of the 16S rRNA gene. For fungi and other eukaryotic mocroorganisms it may be a fragment of the ITS1 (Internally Transcribed Spacer 1) region.

These regions of the genome acquire sequence changes as their organisms evolve over time, such that variants of the sequence can be associated with a particular group of microorganisms, usually at the species or genus level. By amplifying and then sequencing the marker sequence from all suitable organisms in the sample, we can identify the marker sequence variants, and then compare these to databases of known sequences, to identify which organisms are present.

A similar principle applies with _metagenomics_: we sequence the sample and then compare the resulting sequences to a database containing sequences of known organisms, and use this to identify which of them are present in our sample. Differences between the methods arise mainly because this approach is _untargeted_ - we do not know in advance what sequences we will recover - and generates very large amounts of data. Three of the main consequences of this are:

1. We can compare individual sequencing reads to the database of known sequences, but due to the large amount of sequence data (in comparison to metabarcoding) we have to do this using special techniques like _$k$-mer alignment_, rather than direct sequence comparison [@Wood2014-ns].
2. We can assemble near-complete and complete genomes from the data: _Metagenome-Assembled Genomes_ (MAGs) [@Setubal2021-ka].
3. We can obtain much more information than simple organism identity, such as the presence or absence of individual genes or gene functions, such as antimicrobial resistance genes [@De_Abreu2020-jt].

Both _metabarcoding_ and _metagenomics_ can tell us which microorganisms are present in a biological or environmental sample.

### How much of each organism is present?

Whichever method we use to sequence a biological sample, we can obtain a list of the organisms that are present. This tells us something about the composition of the sample, but not everything. There is clearly a difference between one community that is 50% _Escherichia coli_ and 50% _Staphylococcus aureus_, and another community that is 99% _S. aureus_ and 1% _E. coli_. But how can we distinguish between the two?

In both _metabarcoding_ and _metagenomics_ we can count the number of sequences that correspond to each of the organism types we have identified. For metabarcoding, this might be the count of each amplicon sequence corresponding to a microbial genus. For metagenomics, this might be the number of reads assigned to a microbial species by a tool like [Kraken](https://ccb.jhu.edu/software/kraken/).

::: {.callout-warning}
All counts of organisms, by both methods, are strictly _estimates_ of the representation of each organism in the sample. There are many factors that influence the way this estimate may very from the _actual_ amount or proportion of any organism in the sample, including:

- DNA quality [@Manzari2020-br]
- laboratory technique (sample spillover or other handling problems)
- PCR artefacts (especially for low-biomass samples)
- sequence database integrity
- variable accuracy of associating a sequence with an organism or group
:::

With metagenomic analyses that result in assembled MAGs, an additional way of estimating the amount of each organism present is possible: we can count the number of reads that contribute to each MAG assembly. These reads are then assigned the same identity as that of the MAG itself.

We refer to the count of each distinct group of microbes as its _abundance_.

## Describing community composition

Armed with a list of organisms identified as present in our sample, and a (relative) count of how much of the community is composed of each organism, we can describe our measured microbial community composition.
