<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.40">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Leighton Pritchard and Morgan Feeney">

<title>1&nbsp; Microbial Ecology – BM211 Workshops: Online supporting material</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./diversity_measures.html" rel="next">
<link href="./workshop_01_support.html" rel="prev">
<link href="./icon_32.png" rel="icon" type="image/png">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-549806ee2085284f45b00abea8c6df48.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-6c023d2e0400fe6e8bd4ebc2096834a0.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./workshop_01_support.html">Workshop 01 Support</a></li><li class="breadcrumb-item"><a href="./microbial_ecology.html"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Microbial Ecology</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">BM211 Workshops: Online supporting material</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/sipbs-compbiol/BM211-workshops/" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preface to the 2024-25 presentation</span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./workshop_01_support.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Workshop 01 Support</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" role="navigation" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./microbial_ecology.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Microbial Ecology</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./diversity_measures.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Diversity Measures</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./shannon_index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Shannon Index</span></span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#what-is-microbial-ecology" id="toc-what-is-microbial-ecology" class="nav-link active" data-scroll-target="#what-is-microbial-ecology"><span class="header-section-number">1.1</span> What is microbial ecology?</a></li>
  <li><a href="#answering-questions-about-microbial-ecology-using-molecular-techniques" id="toc-answering-questions-about-microbial-ecology-using-molecular-techniques" class="nav-link" data-scroll-target="#answering-questions-about-microbial-ecology-using-molecular-techniques"><span class="header-section-number">1.2</span> Answering questions about microbial ecology using molecular techniques</a>
  <ul class="collapse">
  <li><a href="#which-organisms-are-present" id="toc-which-organisms-are-present" class="nav-link" data-scroll-target="#which-organisms-are-present"><span class="header-section-number">1.2.1</span> Which organisms are present?</a></li>
  <li><a href="#how-much-of-each-organism-is-present" id="toc-how-much-of-each-organism-is-present" class="nav-link" data-scroll-target="#how-much-of-each-organism-is-present"><span class="header-section-number">1.2.2</span> How much of each organism is present?</a></li>
  </ul></li>
  <li><a href="#describing-community-composition" id="toc-describing-community-composition" class="nav-link" data-scroll-target="#describing-community-composition"><span class="header-section-number">1.3</span> Describing community composition</a></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/sipbs-compbiol/BM211-workshops/edit/main/microbial_ecology.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/sipbs-compbiol/BM211-workshops/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default"><nav class="quarto-page-breadcrumbs quarto-title-breadcrumbs d-none d-lg-block" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./workshop_01_support.html">Workshop 01 Support</a></li><li class="breadcrumb-item"><a href="./microbial_ecology.html"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Microbial Ecology</span></a></li></ol></nav>
<div class="quarto-title">
<h1 class="title"><span id="sec-microbial-ecology" class="quarto-section-identifier"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Microbial Ecology</span></span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<p>This section introduces key concepts of how we use molecular techniques to identify and quantify the composition of microbial communities, to study microbial ecology.</p>
<div class="callout callout-style-simple callout-note">
<div class="callout-body d-flex">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-body-container">
<p>This is accompanying material to give additional background and explanation for the workshop. You don’t <em>need</em> to read this for the workshop itself as we will be covering the material in person, but we hope you find it helpful.</p>
</div>
</div>
</div>
<section id="what-is-microbial-ecology" class="level2" data-number="1.1">
<h2 data-number="1.1" class="anchored" data-anchor-id="what-is-microbial-ecology"><span class="header-section-number">1.1</span> What is microbial ecology?</h2>
<p><strong>Microbial ecology is the scientific study of natural microbial communities</strong>, which are critical to many important processes including:</p>
<ul>
<li>airborne microbial communities <span class="citation" data-cites="Santl-Temkiv2022-ou">(<a href="references.html#ref-Santl-Temkiv2022-ou" role="doc-biblioref">Šantl-Temkiv et al. 2022</a>)</span></li>
<li>environmental bioremediation of pollutants <span class="citation" data-cites="Xu2023-fo">(<a href="references.html#ref-Xu2023-fo" role="doc-biblioref">Xu et al. 2023</a>)</span></li>
<li>carbon sequestration in the ocean <span class="citation" data-cites="Castillo2022-hv">(<a href="references.html#ref-Castillo2022-hv" role="doc-biblioref">Castillo et al. 2022</a>)</span></li>
<li>seasonal variation of carbon cycles in soil <span class="citation" data-cites="Poppeliers2022-wn">(<a href="references.html#ref-Poppeliers2022-wn" role="doc-biblioref">Poppeliers et al. 2022</a>)</span></li>
<li>decomposition of animals <span class="citation" data-cites="Mason2023-uy">(<a href="references.html#ref-Mason2023-uy" role="doc-biblioref">Mason, Taylor, and DeBruyn 2023</a>)</span></li>
</ul>
<p>as well as clinical contexts, such as:</p>
<ul>
<li>pathogenesis and nutrient processing in the vertebrate gut <span class="citation" data-cites="Foster-Nyarko2022-xg">(<a href="references.html#ref-Foster-Nyarko2022-xg" role="doc-biblioref">Foster-Nyarko and Pallen 2022</a>)</span></li>
<li>cancer immunotherapy responses <span class="citation" data-cites="Simpson2022-yt">(<a href="references.html#ref-Simpson2022-yt" role="doc-biblioref">Simpson et al. 2022</a>)</span></li>
<li>oral disease <span class="citation" data-cites="Zhang2022-fl">(<a href="references.html#ref-Zhang2022-fl" role="doc-biblioref">Zhang, Whiteley, and Lewin 2022</a>)</span></li>
<li>infant health <span class="citation" data-cites="Qi2022-lz">(<a href="references.html#ref-Qi2022-lz" role="doc-biblioref">Qi et al. 2022</a>)</span></li>
</ul>
<p>It is a hugely important, but arguably underdeveloped area of biology. Important questions remain to be completely answered in detail for many biological systems:</p>
<div class="callout callout-style-simple callout-tip callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-1-contents" aria-controls="callout-1" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
The composition of microbial communities
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-1" class="callout-1-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<ul>
<li>what microbes are found in a community?</li>
<li>what proportion of each microbe species (or other group) is present in a community?</li>
<li>what are the physical limits of the community?</li>
</ul>
</div>
</div>
</div>
<div class="callout callout-style-simple callout-tip callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-2-contents" aria-controls="callout-2" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
interactions within the community
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-2" class="callout-2-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<ul>
<li>is the community <em>stable</em> (or is it growing, shrinking, or changing composition)?</li>
<li>do community members share or compete for resources, and which resources?</li>
<li>do the community members effectively operate as a larger-scale system?</li>
</ul>
</div>
</div>
</div>
<div class="callout callout-style-simple callout-tip callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-3-contents" aria-controls="callout-3" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
interactions of a community with other organisms
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-3" class="callout-3-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<ul>
<li>does the gut microbial community influence the host animal’s nutrition, health, or disease status?</li>
<li>does the rhizosphere microbial community influence a plant’s ability to extract nutrition or energy from the environment, and contribute to protection against pathogens?</li>
</ul>
</div>
</div>
</div>
<div class="callout callout-style-simple callout-tip callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-4-contents" aria-controls="callout-4" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Interactions between communities
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-4" class="callout-4-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<ul>
<li>when two communities come together, what does the resulting community look like?</li>
<li>are there two or more separate (or interacting communities), or do they somehow combine?</li>
</ul>
</div>
</div>
</div>
<div class="callout callout-style-simple callout-tip callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-5-contents" aria-controls="callout-5" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Engineering new microbial communities
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-5" class="callout-5-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<ul>
<li>can we predict which microbes can combine into a productive community?</li>
<li>can we choose these microbes to engineer a community that achieves a particular goal (e.g.&nbsp;plant protection or gut health)</li>
<li>can we perturb existing communities to make them beneficial to health or achieve some other goal?</li>
</ul>
</div>
</div>
</div>
<p>This array of questions can be summarised more simply as <span class="citation" data-cites="Prosser2020-aw">(<a href="references.html#ref-Prosser2020-aw" role="doc-biblioref">Prosser 2020</a>)</span>:</p>
<ol type="1">
<li>Who is there?</li>
<li>What is meant by “there”?</li>
<li>Who is doing what in there?</li>
<li>What is the effect of doing … to “there”?</li>
</ol>
</section>
<section id="answering-questions-about-microbial-ecology-using-molecular-techniques" class="level2" data-number="1.2">
<h2 data-number="1.2" class="anchored" data-anchor-id="answering-questions-about-microbial-ecology-using-molecular-techniques"><span class="header-section-number">1.2</span> Answering questions about microbial ecology using molecular techniques</h2>
<p>Molecular techniques enabled by modern sequencing approaches, such as amplicon marker sequencing (e.g.&nbsp;16S, ITS1 methods) - also known as <strong>metabarcoding</strong> - and sequencing the entire genomic DNA of a complex sample - also known as <strong>metagenomics</strong> - have become the dominant approach to understanding microbial community composition <span class="citation" data-cites="Boughner2016-vx">(<a href="references.html#ref-Boughner2016-vx" role="doc-biblioref">Boughner and Singh 2016</a>)</span>. These tools give rapid insights into the complex composition of microbial communities that are unattainable using laborious, time-consuming culture-based techniques.</p>
<div class="callout callout-style-simple callout-caution callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-6-contents" aria-controls="callout-6" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Metagenomics <em>vs</em> Metabarcoding
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-6" class="callout-6-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>The two words are quite similar, but it is important to distinguish between</p>
<ul>
<li><strong>metabarcoding</strong>: sequencing a single marker sequence from a population</li>
<li><strong>metagenomics</strong>: deep sequencing of the entire genomic DNA material in a sample, often attempting to assemble representative genomes from the data</li>
</ul>
</div>
</div>
</div>
<p>In addition to their speed and the scale of data they can produce, these molecular techniques have other advantages:</p>
<ul>
<li>The (sequence) data collected can be preserved and shared, alongside metadata describing the experiment, in public databases. This enables reanalysis and integration into larger-scale studies.</li>
<li>Fastidious organisms, or those that might be outcompeted in a laboratory culture, can be sequenced and identified.</li>
<li>Organisms present in low numbers can be sequenced and identified.</li>
</ul>
<div class="callout callout-style-simple callout-important">
<div class="callout-body d-flex">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-body-container">
<p>These molecular techniques answer the question of <em>“Who is there?”</em>, in two ways:</p>
<ol type="1">
<li>They tell us <strong>what kinds of organism are present</strong> (often to <em>species</em> or <em>genus</em> level)</li>
<li>They give us a relative count of <strong>how much of each kind of organism is present</strong></li>
</ol>
<p>Taken together, these answers tell us the <strong>diversity of the community</strong>.</p>
</div>
</div>
</div>
<div class="callout callout-style-simple callout-note">
<div class="callout-body d-flex">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-body-container">
<p>We will introduce measures of community diversity in <a href="./diversity_measures.html">a later section</a>.</p>
</div>
</div>
</div>
<section id="which-organisms-are-present" class="level3" data-number="1.2.1">
<h3 data-number="1.2.1" class="anchored" data-anchor-id="which-organisms-are-present"><span class="header-section-number">1.2.1</span> Which organisms are present?</h3>
<p>Both <strong>metabarcoding</strong> and <strong>metagenomics</strong> can tell us which microorganisms are present in a biological or environmental sample.</p>
<section id="metabarcoding" class="level4" data-number="1.2.1.1">
<h4 data-number="1.2.1.1" class="anchored" data-anchor-id="metabarcoding"><span class="header-section-number">1.2.1.1</span> Metabarcoding</h4>
<p>In <strong>metabarcoding</strong>, we use the polymerase chain reaction (PCR) and <strong>primers targeted specifically to amplify a marker sequence</strong>: a gene fragment or other stretch of DNA that we believe to be present in all organisms of interest. For bacteria, this is usually a fragment of the 16S rRNA gene. For fungi and other eukaryotic mocroorganisms it may be a fragment of the ITS1 (Internally Transcribed Spacer 1) region.</p>
<p>These regions of the genome acquire sequence changes as their organisms evolve over time, such that variants of the marker sequence can be associated with a particular group of microorganisms, usually at the species or genus level. By amplifying and then sequencing the marker sequence from all suitable organisms in the sample, we can <strong>identify the marker sequence variants, and then compare these to databases of known sequences</strong>, to identify which organisms are present.</p>
</section>
<section id="metagenomics" class="level4" data-number="1.2.1.2">
<h4 data-number="1.2.1.2" class="anchored" data-anchor-id="metagenomics"><span class="header-section-number">1.2.1.2</span> Metagenomics</h4>
<p>A similar principle applies with <strong>metagenomics</strong>: we sequence the sample and then compare the resulting sequences to a database containing sequences of known organisms, and use this to identify which of them are present in our sample. Differences between the methods arise mainly because <strong>this approach is untargeted - we do not know in advance what sequences we will recover</strong> - and generates very large amounts of data. Three of the main consequences of this are:</p>
<ol type="1">
<li>We can compare individual sequencing reads to the database of known sequences, but due to the large amount of sequence data (in comparison to metabarcoding) we have to do this using special techniques like <strong><span class="math inline">\(k\)</span>-mer alignment</strong>, rather than direct sequence comparison <span class="citation" data-cites="Wood2014-ns">(<a href="references.html#ref-Wood2014-ns" role="doc-biblioref">Wood and Salzberg 2014</a>)</span>.</li>
<li>We can assemble near-complete and complete genomes from the data: <strong>Metagenome-Assembled Genomes (MAGs).</strong> <span class="citation" data-cites="Setubal2021-ka">(<a href="references.html#ref-Setubal2021-ka" role="doc-biblioref">Setubal 2021</a>)</span>.</li>
<li>We can obtain much more information than simple organism identity, such as the presence or absence of individual genes or gene functions, such as antimicrobial resistance genes <span class="citation" data-cites="De_Abreu2020-jt">(<a href="references.html#ref-De_Abreu2020-jt" role="doc-biblioref">Abreu, Perdigão, and Almeida 2020</a>)</span>.</li>
</ol>
</section>
</section>
<section id="how-much-of-each-organism-is-present" class="level3" data-number="1.2.2">
<h3 data-number="1.2.2" class="anchored" data-anchor-id="how-much-of-each-organism-is-present"><span class="header-section-number">1.2.2</span> How much of each organism is present?</h3>
<p>Whichever method we use to sequence a biological sample, we can obtain a list of the organisms that are present. This tells us something about the composition of the sample, but not everything. There is clearly a difference between one community that is 50% <em>Escherichia coli</em> and 50% <em>Staphylococcus aureus</em>, and another community that is 99% <em>S. aureus</em> and 1% <em>E. coli</em>. But how can we distinguish between the two?</p>
<p>In both metabarcoding and metagenomics we can count the number of sequences that correspond to each of the organism types we have identified. For metabarcoding, this might be the count of each amplicon sequence corresponding to a microbial genus. For metagenomics, this might be the number of reads assigned to a microbial species by a tool like <a href="https://ccb.jhu.edu/software/kraken/">Kraken</a>.</p>
<div class="callout callout-style-simple callout-warning callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-7-contents" aria-controls="callout-7" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Molecular techniques only <em>estimate</em> organism abundance
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-7" class="callout-7-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>All counts of organisms, by both methods, are strictly <strong>estimates of the representation of each organism in the sample</strong>. There are many factors that influence the way this estimate may very from the actual amount or proportion of any organism in the sample, including:</p>
<ul>
<li>DNA quality <span class="citation" data-cites="Manzari2020-br">(<a href="references.html#ref-Manzari2020-br" role="doc-biblioref">Manzari et al. 2020</a>)</span></li>
<li>laboratory technique (sample spillover or other handling problems)</li>
<li>PCR artefacts (especially for low-biomass samples)</li>
<li>sequence database integrity</li>
<li>variable accuracy of associating a sequence with an organism or group (taxonomic misassignment)</li>
</ul>
</div>
</div>
</div>
<p>With metagenomic analyses that result in assembled MAGs, an additional way of estimating the amount of each organism present is possible: we can count the number of reads that contribute to each MAG assembly. These reads are then assigned the same identity as that of the MAG itself.</p>
<p>We refer to the count of each distinct group of microbes as its <strong>abundance</strong>.</p>
</section>
</section>
<section id="describing-community-composition" class="level2" data-number="1.3">
<h2 data-number="1.3" class="anchored" data-anchor-id="describing-community-composition"><span class="header-section-number">1.3</span> Describing community composition</h2>
<p>The list of organisms identified as present in our sample and the (relative) count of how much of the community is composed of each organism, taken together, describe our measured <strong>microbial community composition</strong>.</p>


<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list" style="display: none">
<div id="ref-De_Abreu2020-jt" class="csl-entry" role="listitem">
Abreu, Vinicius A C de, José Perdigão, and Sintia Almeida. 2020. <span>“Metagenomic Approaches to Analyze Antimicrobial Resistance: An Overview.”</span> <em>Front. Genet.</em> 11: 575592. <a href="https://doi.org/10.3389%2Ffgene.2020.575592">https://doi.org/10.3389%2Ffgene.2020.575592</a>.
</div>
<div id="ref-Boughner2016-vx" class="csl-entry" role="listitem">
Boughner, Lisa A, and Pallavi Singh. 2016. <span>“Microbial Ecology: Where Are We Now?”</span> <em>Postdoc J.</em> 4 (11): 3–17. <a href="https://doi.org/10.14304%2FSURYA.JPR.V4N11.2">https://doi.org/10.14304%2FSURYA.JPR.V4N11.2</a>.
</div>
<div id="ref-Castillo2022-hv" class="csl-entry" role="listitem">
Castillo, Diego J, Choaro D Dithugoe, Oliver K Bezuidt, and Thulani P Makhalanyane. 2022. <span>“Microbial Ecology of the Southern Ocean.”</span> <em>FEMS Microbiol. Ecol.</em> 98 (11). <a href="https://doi.org/10.1093/femsec/fiac123">https://doi.org/10.1093/femsec/fiac123</a>.
</div>
<div id="ref-Foster-Nyarko2022-xg" class="csl-entry" role="listitem">
Foster-Nyarko, Ebenezer, and Mark J Pallen. 2022. <span>“The Microbial Ecology of Escherichia Coli in the Vertebrate Gut.”</span> <em>FEMS Microbiol. Rev.</em> 46 (3). <a href="https://doi.org/10.1093/femsre/fuac008">https://doi.org/10.1093/femsre/fuac008</a>.
</div>
<div id="ref-Manzari2020-br" class="csl-entry" role="listitem">
Manzari, Caterina, Annarita Oranger, Bruno Fosso, Elisabetta Piancone, Graziano Pesole, and Anna Maria D’Erchia. 2020. <span>“Accurate Quantification of Bacterial Abundance in Metagenomic <span>DNAs</span> Accounting for Variable <span>DNA</span> Integrity Levels.”</span> <em>Microb. Genom.</em> 6 (10). <a href="https://doi.org/10.1099%2Fmgen.0.000417">https://doi.org/10.1099%2Fmgen.0.000417</a>.
</div>
<div id="ref-Mason2023-uy" class="csl-entry" role="listitem">
Mason, Allison R, Lois S Taylor, and Jennifer M DeBruyn. 2023. <span>“Microbial Ecology of Vertebrate Decomposition in Terrestrial Ecosystems.”</span> <em>FEMS Microbiol. Ecol.</em> 99 (2). <a href="https://doi.org/10.1093/femsec/fiad006">https://doi.org/10.1093/femsec/fiad006</a>.
</div>
<div id="ref-Poppeliers2022-wn" class="csl-entry" role="listitem">
Poppeliers, Sanne W M, Mariet Hefting, Ellen Dorrepaal, and James T Weedon. 2022. <span>“Functional Microbial Ecology in Arctic Soils: The Need for a Year-Round Perspective.”</span> <em>FEMS Microbiol. Ecol.</em> 98 (12). <a href="https://doi.org/10.1093/femsec/fiac134">https://doi.org/10.1093/femsec/fiac134</a>.
</div>
<div id="ref-Prosser2020-aw" class="csl-entry" role="listitem">
Prosser, James I. 2020. <span>“Putting Science Back into Microbial Ecology: A Question of Approach.”</span> <em>Philos. Trans. R. Soc. Lond. B Biol. Sci.</em> 375 (1798): 20190240. <a href="https://doi.org/10.1098/rstb.2019.0240">https://doi.org/10.1098/rstb.2019.0240</a>.
</div>
<div id="ref-Qi2022-lz" class="csl-entry" role="listitem">
Qi, Qi, Liang Wang, Mitslal Abrha Gebremedhin, Shaoru Li, Xueyao Wang, Jiali Shen, Yingze Zhu, et al. 2022. <span>“The Impact of Early-Life Antibiotics and Probiotics on Gut Microbial Ecology and Infant Health Outcomes: A Pregnancy and Birth Cohort in Northwest China (<span>PBCC</span>) Study Protocol.”</span> <em>BMC Pediatr.</em> 22 (1): 738. <a href="https://doi.org/10.1186/s12887-022-03811-3">https://doi.org/10.1186/s12887-022-03811-3</a>.
</div>
<div id="ref-Santl-Temkiv2022-ou" class="csl-entry" role="listitem">
Šantl-Temkiv, Tina, Pierre Amato, Emilio O Casamayor, Patrick K H Lee, and Stephen B Pointing. 2022. <span>“Microbial Ecology of the Atmosphere.”</span> <em>FEMS Microbiol. Rev.</em> 46 (4). <a href="https://doi.org/10.1093/femsre/fuac009">https://doi.org/10.1093/femsre/fuac009</a>.
</div>
<div id="ref-Setubal2021-ka" class="csl-entry" role="listitem">
Setubal, João C. 2021. <span>“Metagenome-Assembled Genomes: Concepts, Analogies, and Challenges.”</span> <em>Biophys. Rev.</em> 13 (6): 905–9. <a href="https://doi.org/10.1007/s12551-021-00865-y">https://doi.org/10.1007/s12551-021-00865-y</a>.
</div>
<div id="ref-Simpson2022-yt" class="csl-entry" role="listitem">
Simpson, Rebecca C, Erin R Shanahan, Marcel Batten, Irene L M Reijers, Mark Read, Ines P Silva, Judith M Versluis, et al. 2022. <span>“Diet-Driven Microbial Ecology Underpins Associations Between Cancer Immunotherapy Outcomes and the Gut Microbiome.”</span> <em>Nat. Med.</em> 28 (11): 2344–52. <a href="https://doi.org/10.1038/s41591-022-01965-2">https://doi.org/10.1038/s41591-022-01965-2</a>.
</div>
<div id="ref-Wood2014-ns" class="csl-entry" role="listitem">
Wood, Derrick E, and Steven L Salzberg. 2014. <span>“Kraken: Ultrafast Metagenomic Sequence Classification Using Exact Alignments.”</span> <em>Genome Biol.</em> 15 (3): R46. <a href="https://doi.org/10.1186/gb-2014-15-3-r46">https://doi.org/10.1186/gb-2014-15-3-r46</a>.
</div>
<div id="ref-Xu2023-fo" class="csl-entry" role="listitem">
Xu, Guofang, Siyan Zhao, Jinting Liu, and Jianzhong He. 2023. <span>“Bioremediation of Organohalide Pollutants: Progress, Microbial Ecology, and Emerging Computational Tools.”</span> <em>Curr. Opin. Environ. Sci. Health</em> 32 (100452): 100452. <a href="https://doi.org/10.1016/j.coesh.2023.100452">https://doi.org/10.1016/j.coesh.2023.100452</a>.
</div>
<div id="ref-Zhang2022-fl" class="csl-entry" role="listitem">
Zhang, Mengshi, Marvin Whiteley, and Gina R Lewin. 2022. <span>“Polymicrobial Interactions of Oral Microbiota: A Historical Review and Current Perspective.”</span> <em>MBio</em> 13 (3): e0023522. <a href="https://doi.org/10.1128/mbio.00235-22">https://doi.org/10.1128/mbio.00235-22</a>.
</div>
</div>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./workshop_01_support.html" class="pagination-link" aria-label="Workshop 01 Support">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Workshop 01 Support</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./diversity_measures.html" class="pagination-link" aria-label="Diversity Measures">
        <span class="nav-page-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Diversity Measures</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>These BM211 workshop materials were written by Leighton Pritchard and Morgan Feeney.</p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    <div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/sipbs-compbiol/BM211-workshops/edit/main/microbial_ecology.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/sipbs-compbiol/BM211-workshops/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>This book was built with <a href="https://quarto.org/">Quarto</a>.</p>
</div>
  </div>
</footer>




</body></html>