<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.3.450">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Leighton Pritchard and Morgan Feeney">

<title>bm211-workshops - 3&nbsp; Microbial Ecology</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./diversity_measures.html" rel="next">
<link href="./workshop_01.html" rel="prev">
<link href="./cover.png" rel="icon" type="image/png">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
      <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./workshop_01.html">Workshop 01</a></li><li class="breadcrumb-item"><a href="./microbial_ecology.html"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Microbial Ecology</span></a></li></ol></nav>
      <a class="flex-grow-1" role="button" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
      </a>
      <button type="button" class="btn quarto-search-button" aria-label="" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">bm211-workshops</a> 
        <div class="sidebar-tools-main">
    <a href="https://github.com/sipbs-compbiol/bm211-workshops/" rel="" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
  <a href="" class="quarto-reader-toggle quarto-navigation-tool px-1" onclick="window.quartoToggleReader(); return false;" title="Toggle reader mode">
  <div class="quarto-reader-toggle-btn">
  <i class="bi"></i>
  </div>
</a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Preface to the 2023-24 presentation</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./summary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Summary</span></span></a>
  </div>
</li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./workshop_01.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Workshop 01</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-1" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-1" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./microbial_ecology.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Microbial Ecology</span></span></a>
  </div>
</li>
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./diversity_measures.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Diversity Measures</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./workshop_04.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Workshop 04</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-2" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-2" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./piler_cr.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">PILER-CR</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item sidebar-item-section">
      <div class="sidebar-item-container"> 
            <a href="./workshop_05.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Workshop 05</span></a>
          <a class="sidebar-item-toggle text-start" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar-section-3" aria-expanded="true" aria-label="Toggle section">
            <i class="bi bi-chevron-right ms-2"></i>
          </a> 
      </div>
      <ul id="quarto-sidebar-section-3" class="collapse list-unstyled sidebar-section depth1 show">  
          <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./protein_trees.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Protein Trees</span></a>
  </div>
</li>
      </ul>
  </li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" data-bs-toggle="collapse" data-bs-target="#quarto-sidebar,#quarto-sidebar-glass"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#what-is-microbial-ecology" id="toc-what-is-microbial-ecology" class="nav-link active" data-scroll-target="#what-is-microbial-ecology"><span class="header-section-number">3.1</span> What is microbial ecology?</a></li>
  <li><a href="#answering-questions-about-microbial-ecology-using-molecular-techniques" id="toc-answering-questions-about-microbial-ecology-using-molecular-techniques" class="nav-link" data-scroll-target="#answering-questions-about-microbial-ecology-using-molecular-techniques"><span class="header-section-number">3.2</span> Answering questions about microbial ecology using molecular techniques</a>
  <ul class="collapse">
  <li><a href="#which-organisms-are-present" id="toc-which-organisms-are-present" class="nav-link" data-scroll-target="#which-organisms-are-present"><span class="header-section-number">3.2.1</span> Which organisms are present?</a></li>
  <li><a href="#how-much-of-each-organism-is-present" id="toc-how-much-of-each-organism-is-present" class="nav-link" data-scroll-target="#how-much-of-each-organism-is-present"><span class="header-section-number">3.2.2</span> How much of each organism is present?</a></li>
  </ul></li>
  <li><a href="#describing-community-composition" id="toc-describing-community-composition" class="nav-link" data-scroll-target="#describing-community-composition"><span class="header-section-number">3.3</span> Describing community composition</a></li>
  </ul>
<div class="toc-actions"><div><i class="bi bi-github"></i></div><div class="action-links"><p><a href="https://github.com/sipbs-compbiol/bm211-workshops/edit/main/microbial_ecology.qmd" class="toc-action">Edit this page</a></p><p><a href="https://github.com/sipbs-compbiol/bm211-workshops/issues/new" class="toc-action">Report an issue</a></p></div></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span id="sec-microbial-ecology" class="quarto-section-identifier"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Microbial Ecology</span></span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  

</header>

<p>This section introduces key concepts of how we use molecular techniques to identify and quantify the composition of microbial communities, to study microbial ecology.</p>
<section id="what-is-microbial-ecology" class="level2" data-number="3.1">
<h2 data-number="3.1" class="anchored" data-anchor-id="what-is-microbial-ecology"><span class="header-section-number">3.1</span> What is microbial ecology?</h2>
<p>Microbial ecology is the scientific study of natural microbial communities. It is a hugely important, but arguably underdeveloped area of biology. Many important questions remain to be completely answered in detail:</p>
<ul>
<li>the composition of microbial communities
<ul>
<li>what microbes are found in a community?</li>
<li>what proportion of each microbe species (or other group) is present in a community?</li>
<li>what are the physical limits of the community?</li>
</ul></li>
<li>interactions within the community
<ul>
<li>is the community <em>stable</em> (or is it growing, shrinking, or changing composition)?</li>
<li>do community members share or compete for resources, and which resources?</li>
<li>do the community members effectively operate as a larger-scale system?</li>
</ul></li>
<li>interactions of a community with other organisms
<ul>
<li>does the gut microbial community influence the host animal’s nutrition, health, or disease status?</li>
<li>does the rhizosphere microbial community influence a plant’s ability to extract nutrition or energy from the environment, and contribute to protection against pathogens?</li>
</ul></li>
<li>interactions between communities
<ul>
<li>when two communities come together, what does the resulting community look like?</li>
<li>are there two or more separate (or interacting communities), or do they somehow combine?</li>
</ul></li>
<li>engineering new microbial communities
<ul>
<li>can we predict which microbes can combine into a productive community?</li>
<li>can we choose these microbes to engineer a community that achieves a particular goal (e.g.&nbsp;plant protection or gut health)</li>
<li>can we perturb existing communities to make them beneficial to health or achieve some other goal?</li>
</ul></li>
</ul>
<p>This array of questions can be summarised more simply as <span class="citation" data-cites="Prosser2020-aw">(<a href="references.html#ref-Prosser2020-aw" role="doc-biblioref">Prosser 2020</a>)</span>:</p>
<ol type="1">
<li>Who is there?</li>
<li>What is meant by “there”?</li>
<li>Who is doing what in there?</li>
<li>What is the effect of doing … to there?</li>
</ol>
</section>
<section id="answering-questions-about-microbial-ecology-using-molecular-techniques" class="level2" data-number="3.2">
<h2 data-number="3.2" class="anchored" data-anchor-id="answering-questions-about-microbial-ecology-using-molecular-techniques"><span class="header-section-number">3.2</span> Answering questions about microbial ecology using molecular techniques</h2>
<p>Molecular techniques enabled by modern sequencing approaches, such as amplicon marker sequencing (e.g.&nbsp;16S, ITS1 methods) - also known as <em>metabarcoding</em> - and sequencing the entire genomic DNA of a complex sample - also known as <em>metagenomics</em> - have become the dominant approach to understanding microbial community composition <span class="citation" data-cites="Boughner2016-vx">(<a href="references.html#ref-Boughner2016-vx" role="doc-biblioref">Boughner and Singh 2016</a>)</span>. These tools give rapid insights into the complex composition of microbial communities that are unattainable using laborious, time-consuming culture-based techniques.</p>
<div class="callout callout-style-simple callout-tip">
<div class="callout-body d-flex">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-body-container">
<p>The two words are quite similar, but it is important to distinguish between</p>
<ul>
<li><strong>metabarcoding</strong>: sequencing a single marker sequence from a population</li>
<li><strong>metagenomics</strong>: deep sequencing of the entire genomic DNA material in a sample, often attempting to assemble representative genomes from the data</li>
</ul>
</div>
</div>
</div>
<p>In addition to their speed and the scale of data they can produce, these molecular techniques have additional advantages:</p>
<ul>
<li>The (sequence) data collected can be preserved and shared, alongside metadata describing the experiment, in public databases. This enables reanalysis and integration into larger-scale studies.</li>
<li>Fastidious organisms, or those that might be outcompeted in a laboratory culture, can be sequenced and identified.</li>
<li>Organisms present in low numbers can be sequenced and identified.</li>
</ul>
<p>These molecular techniques answer the question of <em>“Who is there?”</em>, in two ways:</p>
<ol type="1">
<li>They tell us what kinds of organism are present (often to <em>species</em> or <em>genus</em> level)</li>
<li>They give us a relative count of how much of each kind of organism is present</li>
</ol>
<p>Taken together, these answers tell us about the <em>diversity</em> of the community.</p>
<div class="callout callout-style-simple callout-note">
<div class="callout-body d-flex">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-body-container">
<p>We will introduce measures of community diversity in <a href="./diversity_measures.html">a later section</a>.</p>
</div>
</div>
</div>
<section id="which-organisms-are-present" class="level3" data-number="3.2.1">
<h3 data-number="3.2.1" class="anchored" data-anchor-id="which-organisms-are-present"><span class="header-section-number">3.2.1</span> Which organisms are present?</h3>
<p>The two approaches of <em>metabarcoding</em> and <em>metagenomics</em> can each tell us which organisms are present in a sample.</p>
<p>In <em>metabarcoding</em>, we use the polymerase chain reaction (PCR) and primers <em>targeted</em> specifically to amplify a gene fragment or other stretch of DNA that we believe to be present in all organisms of interest. For bacteria, this is usually a fragment of the 16S rRNA gene. For fungi and other eukaryotic mocroorganisms it may be a fragment of the ITS1 (Internally Transcribed Spacer 1) region.</p>
<p>These regions of the genome acquire sequence changes as their organisms evolve over time, such that variants of the sequence can be associated with a particular group of microorganisms, usually at the species or genus level. By amplifying and then sequencing the marker sequence from all suitable organisms in the sample, we can identify the marker sequence variants, and then compare these to databases of known sequences, to identify which organisms are present.</p>
<p>A similar principle applies with <em>metagenomics</em>: we sequence the sample and then compare the resulting sequences to a database containing sequences of known organisms, and use this to identify which of them are present in our sample. Differences between the methods arise mainly because this approach is <em>untargeted</em> - we do not know in advance what sequences we will recover - and generates very large amounts of data. Three of the main consequences of this are:</p>
<ol type="1">
<li>We can compare individual sequencing reads to the database of known sequences, but due to the large amount of sequence data (in comparison to metabarcoding) we have to do this using special techniques like <em><span class="math inline">\(k\)</span>-mer alignment</em>, rather than direct sequence comparison <span class="citation" data-cites="Wood2014-ns">(<a href="references.html#ref-Wood2014-ns" role="doc-biblioref">Wood and Salzberg 2014</a>)</span>.</li>
<li>We can assemble near-complete and complete genomes from the data: <em>Metagenome-Assembled Genomes</em> (MAGs) <span class="citation" data-cites="Setubal2021-ka">(<a href="references.html#ref-Setubal2021-ka" role="doc-biblioref">Setubal 2021</a>)</span>.</li>
<li>We can obtain much more information than simple organism identity, such as the presence or absence of individual genes or gene functions, such as antimicrobial resistance genes <span class="citation" data-cites="De_Abreu2020-jt">(<a href="references.html#ref-De_Abreu2020-jt" role="doc-biblioref">Abreu, Perdigão, and Almeida 2020</a>)</span>.</li>
</ol>
<p>Both <em>metabarcoding</em> and <em>metagenomics</em> can tell us which microorganisms are present in a biological or environmental sample.</p>
</section>
<section id="how-much-of-each-organism-is-present" class="level3" data-number="3.2.2">
<h3 data-number="3.2.2" class="anchored" data-anchor-id="how-much-of-each-organism-is-present"><span class="header-section-number">3.2.2</span> How much of each organism is present?</h3>
<p>Whichever method we use to sequence a biological sample, we can obtain a list of the organisms that are present. This tells us something about the composition of the sample, but not everything. There is clearly a difference between one community that is 50% <em>Escherichia coli</em> and 50% <em>Staphylococcus aureus</em>, and another community that is 99% <em>S. aureus</em> and 1% <em>E. coli</em>. But how can we distinguish between the two?</p>
<p>In both <em>metabarcoding</em> and <em>metagenomics</em> we can count the number of sequences that correspond to each of the organism types we have identified. For metabarcoding, this might be the count of each amplicon sequence corresponding to a microbial genus. For metagenomics, this might be the number of reads assigned to a microbial species by a tool like <a href="https://ccb.jhu.edu/software/kraken/">Kraken</a>.</p>
<div class="callout callout-style-simple callout-warning">
<div class="callout-body d-flex">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-body-container">
<p>All counts of organisms, by both methods, are strictly <em>estimates</em> of the representation of each organism in the sample. There are many factors that influence the way this estimate may very from the <em>actual</em> amount or proportion of any organism in the sample, including:</p>
<ul>
<li>DNA quality <span class="citation" data-cites="Manzari2020-br">(<a href="references.html#ref-Manzari2020-br" role="doc-biblioref">Manzari et al. 2020</a>)</span></li>
<li>laboratory technique (sample spillover or other handling problems)</li>
<li>PCR artefacts (especially for low-biomass samples)</li>
<li>sequence database integrity</li>
<li>variable accuracy of associating a sequence with an organism or group</li>
</ul>
</div>
</div>
</div>
<p>With metagenomic analyses that result in assembled MAGs, an additional way of estimating the amount of each organism present is possible: we can count the number of reads that contribute to each MAG assembly. These reads are then assigned the same identity as that of the MAG itself.</p>
<p>We refer to the count of each distinct group of microbes as its <em>abundance</em>.</p>
</section>
</section>
<section id="describing-community-composition" class="level2" data-number="3.3">
<h2 data-number="3.3" class="anchored" data-anchor-id="describing-community-composition"><span class="header-section-number">3.3</span> Describing community composition</h2>
<p>Armed with a list of organisms identified as present in our sample, and a (relative) count of how much of the community is composed of each organism, we can describe our measured microbial community composition.</p>


<div id="refs" class="references csl-bib-body hanging-indent" role="list" style="display: none">
<div id="ref-De_Abreu2020-jt" class="csl-entry" role="listitem">
Abreu, Vinicius A C de, José Perdigão, and Sintia Almeida. 2020. <span>“Metagenomic Approaches to Analyze Antimicrobial Resistance: An Overview.”</span> <em>Front. Genet.</em> 11: 575592. <a href="https://doi.org/10.3389%2Ffgene.2020.575592">https://doi.org/10.3389%2Ffgene.2020.575592</a>.
</div>
<div id="ref-Boughner2016-vx" class="csl-entry" role="listitem">
Boughner, Lisa A, and Pallavi Singh. 2016. <span>“Microbial Ecology: Where Are We Now?”</span> <em>Postdoc J.</em> 4 (11): 3–17. <a href="https://doi.org/10.14304%2FSURYA.JPR.V4N11.2">https://doi.org/10.14304%2FSURYA.JPR.V4N11.2</a>.
</div>
<div id="ref-Manzari2020-br" class="csl-entry" role="listitem">
Manzari, Caterina, Annarita Oranger, Bruno Fosso, Elisabetta Piancone, Graziano Pesole, and Anna Maria D’Erchia. 2020. <span>“Accurate Quantification of Bacterial Abundance in Metagenomic <span>DNAs</span> Accounting for Variable <span>DNA</span> Integrity Levels.”</span> <em>Microb. Genom.</em> 6 (10). <a href="https://doi.org/10.1099%2Fmgen.0.000417">https://doi.org/10.1099%2Fmgen.0.000417</a>.
</div>
<div id="ref-Prosser2020-aw" class="csl-entry" role="listitem">
Prosser, James I. 2020. <span>“Putting Science Back into Microbial Ecology: A Question of Approach.”</span> <em>Philos. Trans. R. Soc. Lond. B Biol. Sci.</em> 375 (1798): 20190240. <a href="https://doi.org/10.1098/rstb.2019.0240">https://doi.org/10.1098/rstb.2019.0240</a>.
</div>
<div id="ref-Setubal2021-ka" class="csl-entry" role="listitem">
Setubal, João C. 2021. <span>“Metagenome-Assembled Genomes: Concepts, Analogies, and Challenges.”</span> <em>Biophys. Rev.</em> 13 (6): 905–9. <a href="https://doi.org/10.1007/s12551-021-00865-y">https://doi.org/10.1007/s12551-021-00865-y</a>.
</div>
<div id="ref-Wood2014-ns" class="csl-entry" role="listitem">
Wood, Derrick E, and Steven L Salzberg. 2014. <span>“Kraken: Ultrafast Metagenomic Sequence Classification Using Exact Alignments.”</span> <em>Genome Biol.</em> 15 (3): R46. <a href="https://doi.org/10.1186/gb-2014-15-3-r46">https://doi.org/10.1186/gb-2014-15-3-r46</a>.
</div>
</div>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./workshop_01.html" class="pagination-link">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text">Workshop 01</span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./diversity_measures.html" class="pagination-link">
        <span class="nav-page-text">Diversity Measures</span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">These BM211 workshops were written by Leighton Pritchard and Morgan Feeney.</div>   
    <div class="nav-footer-center">
      &nbsp;
    </div>
    <div class="nav-footer-right">This book was built with <a href="https://quarto.org/">Quarto</a>.</div>
  </div>
</footer>



</body></html>